### ISSUE TITLE
LWG-3120 Unclear behavior of monotonic_buffer_resource::release()

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#1468]
LWG-3120 Unclear behavior of `monotonic_buffer_resource::release()`

This adds the following Standardese:

> *Effects:* Calls `upstream_rsrc->deallocate()` as necessary to release all allocated memory. **Resets `current_buffer` and `next_buffer_size` to their initial values at construction.**

Our current implementation doesn't appear to do that:

https://github.com/microsoft/STL/blob/9959929c77b18e25b6dfbc8886e311b913321925/stl/inc/memory_resource#L636-L656

**Comment by futuarmo** [comment]
I think we need here in case of increasing allocated memory size to save initial current buffer and next buffer size to member variables, did I get it right?

**Comment by CaseyCarter** [comment]
> I think we need here in case of increasing allocated memory size to save initial current buffer and next buffer size to member variables, did I get it right?

Yes, that's correct, and unfortunately it's also ABI breaking since it will change the size of `monotonic_buffer_resource`.

**Comment by futuarmo** [comment]
@CaseyCarter but can I make PR for fix?
